### PR TITLE
SpectrumInfo and DetectorInfo rollout: algorithms <InstrumentWidgetMaskTab.cpp to QpeaksTableModel.cpp>

### LINKED ISSUE #18616: SpectrumInfo and DetectorInfo rollout: algorithms <ComponentActor.cpp to QpeaksTableModel.cpp>
This issue is part of the SpectrumInfo and DetectorInfo rollout, see #17743.
Algorithms covered in this issue are:

```
MantidQt/MantidWidgets/.../InstrumentView/ComponentActor.cpp
MantidQt/MantidWidgets/.../InstrumentView/InstrumentActor.cpp
MantidQt/MantidWidgets/.../InstrumentView/InstrumentWidgetMaskTab.cpp
MantidQt/MantidWidgets/.../InstrumentView/InstrumentWidgetPickTab.cpp
MantidQt/MantidWidgets/.../InstrumentView/ObjComponentActor.cpp
MantidQt/MantidWidgets/.../InstrumentView/ProjectionSurface.cpp
MantidQt/MantidWidgets/.../InstrumentView/RotationSurface.cpp
```

and 

```
MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/InstrumentView/ComponentActor.h
MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/InstrumentView/InstrumentActor.h
MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/InstrumentView/ProjectionSurface.h
```
and 
```
MantidQt/SliceViewer/.../QpeaksTableModel.cpp
```

### PR DESCRIPTION
**Pull request by AntonPiccardoSelg**
## Description of work.
This PR is part of refactoring effort to use the new interface of the SpectrumInfo and DetectorInfo modules.
See https://github.com/mantidproject/mantid/issues/17743 for details.

The following algorithms and tests were refactored to use SpectrumInfo/DetectorInfo:
```
MantidQt/MantidWidgets/.../InstrumentView/InstrumentWidgetMaskTab.cpp
MantidQt/SliceViewer/.../QpeaksTableModel.cpp
```

Note that all other files mentioned in the issue cannot be converted.

Performance tests were added for:
N/A

The following algorithms have improved performance:
N/A

## To test.

Code review. 

Fixes #18616
Internal change, no release notes (potential performance improvements will be added to release notes as part of the umbrella issue).

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by KarlPalmen** [comment]
:shipit:

File: MantidQt/SliceViewer/src/QPeaksTableModel.cpp [hunk 1]
```diff
@@ -309,14 +310,15 @@ std::vector<int> QPeaksTableModel::defaultHideCols() {
     if (numToCheck > 20) // arbitrary cutoff
       numToCheck = 20;
     const std::string RECT_DET("RectangularDetector");
+    const auto &detectorInfo = m_peaksWS->detectorInfo();
```
Comment (martyngigg) [inline MantidQt/SliceViewer/src/QPeaksTableModel.cpp]: This code doesn't look like it is doing anything at all; even before the changes.

It just breaks out of the loop if it finds a rectangular detector but I can't see how that affects anything. Am I missing something?

Comment (AntonPiccardoSelg) [inline MantidQt/SliceViewer/src/QPeaksTableModel.cpp]: Didn't pay too much attention to the code itself. Yes, it seems that this bit of code is actually pointless. I will remove it.

**Comment by martyngigg** [comment]
@KarlPalmen Can we test the functionality here too and not just code review? Just the functionality affected by `QPeaksTableTableModel` as the other file was just removing code already commented out.

**Comment by KarlPalmen** [comment]
Show detectors and Show data of loaded workspace worked OK.

**Comment by KarlPalmen** [comment]
I expect this document to be helpful in testing http://docs.mantidproject.org/nightly/concepts/PeaksWorkspace.html?highlight=peaksworkspace

**Comment by KarlPalmen** [comment]
The usage example of FindPeaksMD worked for me. I was then able to drag the peaks workspace to the instrument view and it then showed the peaks on the instrument view.

**Comment by martyngigg** [comment]
:+1: Thanks for testing that, Karl.

**Comment by KarlPalmen** [comment]
I then did this with peaks from another usage example and they showed up on the instrument view along with the previous peaks, but in a different colour.

**Pull request merged**